aikido_zen/sinks/shutil.py

```diff
@@ -2,51 +2,32 @@
 Sink module for python's `shutil`
 """
 
-import copy
-import aikido_zen.importhook as importhook
 import aikido_zen.vulnerabilities as vulns
+from aikido_zen.helpers.get_argument import get_argument
+from aikido_zen.sinks import on_import, patch_function, before
 
-# File functions func(src, dst, *, **)
-SHUTIL_SRC_DST_FUNCTIONS = [
-    "copymode",
-    "copystat",
-    "copytree",
-    "move",
-]
-# shutil.copyfile(src, dst, *, **) => builtins.open
-# shutil.copy(src, dst, *, **) => builtins.open
-# shutil.copy2(src, dst, *, **) => builtins.open
 
+@before
+def _shutil_func(func, instance, args, kwargs):
+    source = get_argument(args, kwargs, 0, "src")
+    destination = get_argument(args, kwargs, 1, "dst")
 
-def generate_aikido_function(aik_op, func):
-    """
-    Returns a generated aikido function given an operation
-    and the previous function
-    """
-
-    def wrapper(src, dst, *args, **kwargs):
-        kind = "path_traversal"
-        op = f"shutil.{aik_op}"
-        if src:
-            vulns.run_vulnerability_scan(kind, op, args=(src,))
-        if dst:
-            vulns.run_vulnerability_scan(kind, op, args=(dst,))
-        return func(src, dst, *args, **kwargs)
+    kind = "path_traversal"
+    op = f"shutil.{func.__name__}"
+    if isinstance(source, str):
+        vulns.run_vulnerability_scan(kind, op, args=(source,))
+    if isinstance(source, str):
```

Comment:
```suggestion
    if isinstance(destination, str):
```

Not caught in unit test?